stake-pool/js/test/stakepool-program.test.ts

```diff
@@ -0,0 +1,255 @@
+import {
+  PublicKey,
+  Connection,
+  Keypair,
+  clusterApiUrl,
+  SystemProgram, AccountInfo, LAMPORTS_PER_SOL
+} from '@solana/web3.js';
+import {
+  STAKE_POOL_INSTRUCTION_LAYOUTS,
+  DepositSolParams,
+  StakePoolProgram,
+} from '../src/stakepool-program';
+import { decodeData } from '../src/copied-from-solana-web3/instruction';
+import { depositSol, withdrawSol, withdrawStake } from "../src";
+import { STAKE_POOL_LAYOUT } from "../src/layouts";
+import { mockTokenAccount, mockValidatorList, stakePoolMock } from "./mocks";
+
+describe('StakePoolProgram', () => {
+
+  const connection = new Connection(
+    clusterApiUrl('devnet'),
+    'confirmed',
+  );
+
+  connection.getMinimumBalanceForRentExemption = jest.fn(async () => 10000);
+
+  const STAKE_POOL_PROGRAM_ADDR = new PublicKey(
+    'SPoo1Ku8WFXoNDMHPsrGSTSG1Y47rzgn41SLUNakuHy',
+  );
+
+  const data = Buffer.alloc(1024);
+  STAKE_POOL_LAYOUT.encode(stakePoolMock, data)
+
+  const stakePoolAccount = <AccountInfo<any>>{
+    executable: true,
+    owner: STAKE_POOL_PROGRAM_ADDR,
+    lamports: 99999,
+    data,
+  };
+
+  it('depositSolInstruction', () => {
+
+    const payload: DepositSolParams = {
+      stakePoolPubkey: STAKE_POOL_PROGRAM_ADDR,
```

Comment:
nit: this could be confusing since `stakePoolPubkey` should be the pubkey of a stake pool, and not the pubkey of the program!